bats/tests/containers/switch-engines.bats

```diff
@@ -0,0 +1,54 @@
+# Test case 20
+
+setup() {
+    load '../helpers/load'
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+@test "verify we're on docker" {
+    engineName="$(rdctl list-settings | jq -r .containerEngine.name)"
+    case $engineName in
+    containerd)
+        rdctl set --container-engine.name moby
+        wait_for_container_engine
+        ;;
+    esac
+}
```

Comment:
Since we allow running the test either with `RD_CONTAINER_ENGINE=containerd` or `RD_CONTAINER_ENGINE=moby`, it feels somewhat wrong to force this test to always run with `docker` first.

It should switch `containerd` → `moby` → `containerd` under `RD_CONTAINER_ENGINE=containerd`, and `moby` → `containerd` → `moby` otherwise.

Reply:
was draft, but good point to test both situations